crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -440,14 +440,27 @@ impl Device {
             recipient_key = ?self.curve25519_key(),
             event_type,
             session,
+            message_id,
         ))
     ]
     pub(crate) async fn encrypt(
         &self,
         event_type: &str,
         content: Value,
     ) -> OlmResult<(Session, Raw<ToDeviceEncryptedEventContent>)> {
-        self.inner.encrypt(self.verification_machine.store.inner(), event_type, content).await
+        #[cfg(feature = "message-ids")]
+        let message_id = {
+            let id = ulid::Ulid::new().to_string();
+            tracing::Span::current().record("message_id", &id);
+            Some(id)
+        };
+
+        #[cfg(not(feature = "message-ids"))]
+        let message_id = None;
```

Comment:
out of curiosity, does the instrumented `message_id` need not to be manually recorded in that case?